interfaces/migrate/src/main/java/migrate/interfaces/Migrate.java

```diff
@@ -37,20 +37,21 @@ void migrateSyntax(List<Path> unmanagedSources,
 
     // Todo: Maybe using ServiceLoader could simplify this code a bit:
     // https://www.baeldung.com/java-spi
-    static Migrate fetchAndClassloadInstance(String migrateVersion, String scalaVersion) throws Exception {
+    static Migrate fetchAndClassloadInstance(String migrateVersion, String scalaVersion, Logger logger) throws Exception {
         List<URL> jars = getJars(migrateVersion, scalaVersion);
 
         ClassLoader parent = new MigrateClassloader(Migrate.class.getClassLoader());
         URLClassLoader classLoader = new URLClassLoader(jars.stream().toArray(URL[]::new), parent);
 
-        return classloadInstance(classLoader);
+        return classloadInstance(classLoader, logger);
     }
 
-    static Migrate classloadInstance(URLClassLoader classLoader) throws Exception {
+    static Migrate classloadInstance(URLClassLoader classLoader, Logger logger) throws Exception {
+        Class<?> loggerCls = classLoader.loadClass("migrate.interfaces.Logger");
```

Comment:
This had better be the same as `Logger.class`, since you're passing in something that is statically typed here as `Logger`. Consider directly using `Logger.class` instead.